Src/DependencyCollector/Shared/HttpDependenciesParsingTelemetryInitializer.cs

```diff
@@ -0,0 +1,70 @@
+﻿namespace Microsoft.ApplicationInsights.DependencyCollector
+{
+    using System;
+    using Channel;
+    using DataContracts;
+    using Extensibility;
+
+    /// <summary>
+    /// Telemetry Initializer that parses http dependencies into well-known types like Azure Storage.
+    /// </summary>
+    public class HttpDependenciesParsingTelemetryInitializer : ITelemetryInitializer
+    {
+        /// <summary>
+        /// If telemetry item is http dependency - converts it to the well-known type of the dependency.
+        /// </summary>
+        /// <param name="telemetry">Telemetry item to convert.</param>
+        public void Initialize(ITelemetry telemetry)
+        {
+            var httpDependency = telemetry as DependencyTelemetry;
+
+            if (httpDependency != null && httpDependency.Type != null && httpDependency.Type.ToUpperInvariant() == "HTTP")
+            {
+                string host = httpDependency.Target;
+
+                if (!string.IsNullOrEmpty(host))
+                {
+                    if (host.EndsWith("blob.core.windows.net", StringComparison.OrdinalIgnoreCase))
+                    {
+                        httpDependency.Type = "Azure blob";
+
+                        string nameWithoutVerb = httpDependency.Name;
+                        var verb = GetVerb(httpDependency.Name, out nameWithoutVerb);
+
+                        var isFirstSlash = nameWithoutVerb[0] == '/' ? 1 : 0;
+                        var idx = nameWithoutVerb.IndexOf('/', isFirstSlash); // typically first symbol of the path is '/'
+                        string container = idx != -1 ? nameWithoutVerb.Substring(isFirstSlash, idx - isFirstSlash) : string.Empty;
```

Comment:
What about first two slashes? This code looks suspiciously "breezy" if you know what I mean. Do we have a set of the values generally seen as Name to base our decisions on? Or this is an educated guess?